src/Features/Core/Portable/UseCompoundAssignment/AbstractUseCompoundAssignmentDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,202 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.LanguageServices;
+
+namespace Microsoft.CodeAnalysis.UseCompoundAssignment
+{
+    internal abstract class AbstractUseCompoundAssignmentDiagnosticAnalyzer<
+        TSyntaxKind,
+        TAssignmentSyntax,
+        TBinaryExpressionSyntax>
+        : AbstractCodeStyleDiagnosticAnalyzer
+        where TSyntaxKind : struct
```

Comment:
System.Enum?

Reply:
I don't see that buying us anything here.